pilot/pkg/networking/core/tracing.go

```diff
@@ -247,20 +249,54 @@ func convertTraceContextOption(
 
 func zipkinConfig(
 	hostname, cluster, endpoint string, enable128BitTraceID bool, traceContextOption tracingcfg.ZipkinConfig_TraceContextOption,
-	proxy *model.Proxy,
+	timeout *durationpb.Duration, headers []*meshconfig.MeshConfig_ExtensionProvider_HttpHeader, proxy *model.Proxy,
 ) (*anypb.Any, error) {
 	if endpoint == "" {
 		endpoint = "/api/v2/spans" // envoy deprecated v1 support
 	}
+
+	// Determine if we should use HttpService (modern approach) or legacy fields
+	// For newer proxies (Istio 1.29+), always use HttpService as it's the modern approach
+	// that supports timeout and custom headers. For older proxies, use legacy fields for
+	// backward compatibility.
+	useHTTPService := proxy.IstioVersion != nil && proxy.VersionGreaterOrEqual(&model.IstioVersion{Major: 1, Minor: 29})
```

Comment:
xref: for https://github.com/istio/istio/issues/58399#issuecomment-3577692868 you need to check timeout and headers.